DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py

```diff
@@ -0,0 +1,249 @@
+import DigitalEcosystem.utils.figures
+
+tqdm.tqdm.pandas()
+
+
+# In[]:
+
+
+# Random seeds for reproducibility
+RANDOM_SEED = 1234
+import random
+random.seed(RANDOM_SEED)
+np.random.seed(RANDOM_SEED)
+
+
+# In[]:
+
+
+# Load up the data
+data_path = "../../refined/httpot/full_featurized_data.pkl"
+data = pd.read_pickle(data_path)
+
+cols_to_drop = ['formula',
+                'discovery_process (unitless)',
+                'potcars (unitless)',
+                'is_hubbard (unitless)',
+                'energy_per_atom (eV)',
+                'exfoliation_energy_per_atom (eV/atom)',
+                'is_bandgap_direct (unitless)',
+                'is_metal (unitless)',
+                'energy_vdw_per_atom (eV/atom)',
+                'total_magnetization (Bohr Magneton)']
+target_column = ['exfoliation_energy_per_atom (eV/atom)']
+matpedia_id = ['2dm_id (unitless)']
+atoms_col = ['atoms_object (unitless)']
+
+
+# In[]:
+
+
+noble_gases = ['He', 'Ne', 'Ar', 'Kr', 'Xe', 'Rn']
+fblock = ['La', 'Ce', 'Pr', 'Nd', 'Pm', 'Sm', 'Eu', 'Gd', 'Tb', 'Dy', 'Ho', 'Er', 'Tm', 'Yb', 'Lu',
+          'Ac', 'Th', 'Pa', 'U',  'Np', 'Pu', 'Am', 'Cm', 'Bk', 'Cf', 'Es', 'Fm', 'Md', 'No', 'Lr']
+d_synths = ['Rf', 'Db', 'Sg', 'Bh', 'HS', 'Mt', 'Ds', 'Rg', 'Cn', 'Nh', 'Fl', 'Mc', 'Lv', 'Ts', 'Og']
+
+bad_elements = noble_gases + fblock + d_synths
+
+element_mask = data['atoms_object (unitless)'].apply(lambda atoms: all([forbidden not in atoms.get_chemical_symbols() for forbidden in bad_elements]))
+
+decomp_mask = data['decomposition_energy (eV/atom)'] < 0.5
+
+exfol_mask = data['exfoliation_energy_per_atom (eV/atom)'] > 0
+
+reasonable = data[element_mask & decomp_mask & exfol_mask]
+
+
+# In[]:
+
+
+xenonpy_descriptors = [col for col in data.columns if ":" in col]
+matminer_descriptors = [
+    'bond_length_average',
+    'bond_angle_average',
+    'average_cn',
+    'global_instability',
+    'perimeter_area_ratio',
+    'ewald_energy_per_atom',
+    'structural complexity per atom',
+    'structural complexity per cell',
+    'n_symmetry_ops'
+]
+xenonpy_matminer_descriptors = xenonpy_descriptors + matminer_descriptors
+
+
+# In[]:
+
+
+borderline = np.percentile(reasonable['bandgap (eV)'], 50)
+print(f"50th percentile is at a bandgap of {borderline} eV")
+bandgap_low_mask = reasonable['bandgap (eV)'] <= borderline
+bandgap_high_mask = ~bandgap_low_mask
+
+
+# In[]:
+
+
+reasonable['low_bandgap'] = bandgap_low_mask.astype(int)
+
+
+# In[]:
+
+
+train, test = sklearn.model_selection.train_test_split(reasonable, test_size=0.1, random_state=RANDOM_SEED)
+
+
+# In[]:
+
+
+train_x_reg = np.nan_to_num(train[xenonpy_matminer_descriptors + ['low_bandgap']].to_numpy())
+train_y_reg = np.nan_to_num(train[target_column].to_numpy())
+
+test_x_reg = np.nan_to_num(test[xenonpy_matminer_descriptors + ['low_bandgap']].to_numpy())
+test_y_reg = np.nan_to_num(test[target_column].to_numpy())
+
+
+# In[]:
+
+
+current_reg = None
+best_reg = None
+def keep_best_reg(study, trial):
```

Comment:
Again:

- no shornames
- add explanation

Reply:
Expanded the names and added comments